jetty-server/src/main/java/org/eclipse/jetty/server/AsyncAttributes.java

```diff
@@ -94,54 +83,15 @@ public void setAttribute(String key, Object value)
         switch (key)
         {
             case AsyncContext.ASYNC_REQUEST_URI:
```

Comment:
Seems the intention is to make this immutable. Perhaps we should throw IllegalStateException if setAttribute() is called? That way we would find out where in the code it is being (incorrectly) called, rather than silently failing and being ignored.

Reply:
I don't think ISE is right.   I've change to always setting the attribute on the wrapped map, which we will hide during the forward. I added comments to explain.

Reply:
Why set the attribute on the wrapped map, we never unwrap the AsyncAttributes so this will never be used.

Reply:
Current behaviour is to only set the attribute on the wrapped map for non-async attributes.  Those are used by AsyncAttributes as it passes on non-async gets to the the wrapped map.